lib/code_corps/stripe_service/webhook_processing/webhook_processor.ex

```diff
@@ -5,79 +5,104 @@ defmodule CodeCorps.StripeService.WebhookProcessing.WebhookProcessor do
 
   alias CodeCorps.StripeEvent
   alias CodeCorps.Repo
+  alias CodeCorps.StripeService.WebhookProcessing.{ConnectEventHandler, PlatformEventHandler}
+
+  @api Application.get_env(:code_corps, :stripe)
 
   @doc """
   Used to process a Stripe webhook event.
 
-  Receives the event json as the first parameter.
-  Since a webhook can be a platform or a connect webhook,
-  the function requires the handler module as the second parameter.
+  Receives the event JSON as the first parameter.
+
+  Since a webhook can be a platform or a connect webhook, the function requires
+  the handler module as the second parameter.
 
   ## Returns
 
-  * `{:ok, :ignored_by_environment}` if the event was ignored due to environment mismatch
-  * `{:ok, :enqueued}` if the event will be handled
+  - `{:ok, pid}` if the event will be handled
+  - `{:error, :ignored_by_environment}` if the event was ignored due to
+    environment mismatch
 
   ## Note
 
   Stripe events can have their `livemode` property set to `true` or `false`.
-  A livemode event should be handled by the production environment, while all other environments
-  handle non-livemode events.
+  A livemode `true` event should be handled by the production environment,
+  while all other environments handle livemode `false` events.
   """
-  def process_async(%{} = json, handler) do
-    case event_matches_environment?(json) do
-      true -> do_process_async(json, handler)
-      false -> {:ok, :ignored_by_environment}
+  def process_async(%{"id" => id, "livemode" => livemode, "user_id" => user_id} = json, handler) do
+    case event_matches_environment?(livemode) do
+      true -> do_process_async(id, user_id, handler, json)
+      false -> {:error, :ignored_by_environment}
     end
   end
-
-  defp do_process_async(json, handler) do
-    Task.Supervisor.start_child(:webhook_processor, fn -> do_process(json, handler) end)
+  def process_async(%{"id" => id, "livemode" => livemode} = json, handler) do
```

Comment:
Yah, but isn't it just passing the `user_id` value into the same place where it would be `nil` in the other example? So I thought `json["user_id"]` as the second param to the `true` case would handle both implementations but I could be wrong.